base_layer/core/src/proof_of_work/monero_rx/merkle_tree.rs

```diff
@@ -530,33 +478,11 @@ mod test {
             let expected_root = cn_fast_hash2(&h01, &h2345);
 
             // Proof for h0
-            let proof = create_merkle_proof(&hashes, &hashes[0]).unwrap();
+            let proof = create_merkle_proof(&hashes).unwrap();
             assert_eq!(proof.calculate_root(&hashes[0]), expected_root);
-            assert_eq!(proof.depth, 2);
             assert_eq!(proof.branch().len(), 2);
             assert_eq!(proof.branch()[0], hashes[1]);
             assert_eq!(proof.branch()[1], h2345);
-            assert_eq!(proof.path_bitmap, 0b00000000);
```

Comment:
While checking things I added 2 hashes to this test. Up to you if you want to add it, not NB. Obviously if you do you'll need to update/remove the "diagram" :)

```rust
   let hashes = (1..=8).map(|i| Hash::from([i; 32])).collect::<Vec<_>>();
            let h23 = cn_fast_hash2(&hashes[2], &hashes[3]);
            let h45 = cn_fast_hash2(&hashes[4], &hashes[5]);
            let h67 = cn_fast_hash2(&hashes[6], &hashes[7]);
            let h01 = cn_fast_hash2(&hashes[0], &hashes[1]);
            let h0123 = cn_fast_hash2(&h01, &h23);
            let h4567 = cn_fast_hash2(&h45, &h67);
            let expected_root = cn_fast_hash2(&h0123, &h4567);

            // Proof for h0
            let proof = create_merkle_proof(&hashes).unwrap();
            assert_eq!(proof.calculate_root(&hashes[0]), expected_root);
            assert_eq!(proof.branch().len(), 3);
            assert_eq!(proof.branch()[0], hashes[1]);
            assert_eq!(proof.branch()[1], h23);
            assert_eq!(proof.branch()[2], h4567)
```